### PR TITLE
changed fetch span attributes to be compatible with http

### PR DESCRIPTION
**Pull request by chrome99**
- Removed attributes from fetch that don't exist on http - user_agent, scheme and component.
- Added attributes that are missing on fetch, but exist on http - response_content_length, target, [net.peer.name](http://net.peer.name/).
- Attributes are added to fetch on request or response in the same way as http, and in the same order.
- On error - added attribute error_name and error_message, while also throwing the error event to the events array, so that fetch error handling is identical to http error handling.
- Replaced custom FetchError with regular Error to allow accessing [error.name](http://error.name/).
- Packages: added ava and ts-node for testing.
- Tests: added new test, renamed previous test for ava to ignore it.

### CODE REVIEW

File: experimental/packages/opentelemetry-instrumentation-fetch/test/test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import test from 'ava';
+import { ConsoleSpanExporter, SimpleSpanProcessor, WebTracerProvider } from '@opentelemetry/sdk-trace-web';
+import { FetchInstrumentation } from '../src';
+import { ZoneContextManager } from '@opentelemetry/context-zone';
+import { registerInstrumentations, isWrapped } from '@opentelemetry/instrumentation';
+
+const provider = new WebTracerProvider();
+
+provider.addSpanProcessor(new SimpleSpanProcessor(new ConsoleSpanExporter()));
```
Comment (agamm) [inline experimental/packages/opentelemetry-instrumentation-fetch/test/test.ts:9]: Did you test with Jager too?

Comment (agamm) [reply experimental/packages/opentelemetry-instrumentation-fetch/test/test.ts:9]: (Just a thought)

File: experimental/packages/opentelemetry-instrumentation-fetch/src/fetch.ts [hunk 2]
```diff
@@ -270,6 +314,23 @@ export class FetchInstrumentation extends InstrumentationBase<
     this._usedResources.add(resource);
   }
 
+  /**
+   * Sets the span with the error passed in params
+   * @param span the span that need to be set
+   * @param error error that will be set to span
+   */
+  private _handleSpanWithError(span: api.Span, error: Error): void {
```
Comment (agamm) [inline experimental/packages/opentelemetry-instrumentation-fetch/src/fetch.ts:322]: 👍

**Review by agamm — APPROVED** [review]
Briefly looked, nice.

**Comment by agamm** [comment]
I would keep the User agent and scheme (ask ben to double check).

File: experimental/packages/opentelemetry-instrumentation-fetch/test/test.ts (hunk 1 above)
Comment (chrome99) [reply experimental/packages/opentelemetry-instrumentation-fetch/test/test.ts:9]: Yes, Now I tried this also with Jaeger!

**Comment by chrome99** [comment]
> I would keep the User agent and scheme (ask ben to double check).

I asked Ben, waiting for his reply.

**Pull request merged**